src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -1408,35 +1409,41 @@ public void readExternal(DataInputStream dis, PrototypeFactory pf) throws IOExce
      *                            (presumably in HQ) - so don't fire end of form event.
      */
     public void initialize(boolean newInstance, boolean isCompletedInstance,
-                           InstanceInitializationFactory factory) {
-        initialize(newInstance, isCompletedInstance, factory, null, false);
+                           InstanceInitializationFactory factory, CommCareSession commCareSession) {
+        initialize(newInstance, isCompletedInstance, factory, null, false, commCareSession);
     }
 
-    public void initialize(boolean newInstance, InstanceInitializationFactory factory) {
-        initialize(newInstance, false, factory, null, false);
+    public void initialize(boolean newInstance, InstanceInitializationFactory factory, CommCareSession commCareSession) {
+        initialize(newInstance, false, factory, null, false, commCareSession);
     }
 
     public void initialize(boolean newInstance, InstanceInitializationFactory factory, String locale,
-                           boolean isReadOnly) {
-        initialize(newInstance, false, factory, locale, isReadOnly);
+                           boolean isReadOnly, CommCareSession commCareSession) {
+        initialize(newInstance, false, factory, locale, isReadOnly, commCareSession);
     }
 
     /**
      * meant to be called after deserialization and initialization of handlers
      *
-     * @param newInstance true if the form is to be used for a new entry interaction,
-     *                    false if it is using an existing IDataModel
-     * @param locale      The default locale in the current environment, if provided. Can be null
-     *                    to rely on the form's internal default.
+     * @param newInstance     true if the form is to be used for a new entry interaction,
+     *                        false if it is using an existing IDataModel
+     * @param locale          The default locale in the current environment, if provided. Can be null
+     *                        to rely on the form's internal default.
+     * @param isReadOnly      If we are in read only mode and only wants to view form
+     * @param commCareSession Current commcare session the form is opened in
      */
     public void initialize(boolean newInstance, boolean isCompletedInstance,
-                           InstanceInitializationFactory factory, String locale, boolean isReadOnly) {
+                           InstanceInitializationFactory factory, String locale, boolean isReadOnly, CommCareSession commCareSession) {
```

Comment:
Why aren't the instances from the session here coming through the IIF? The purpose of that object is to be able to provide instance initializers from different frames of context, so I think we should be relying on it rather than extending initialization to include a new coupled dependency.

Reply:
Thanks this was a mix up on my part. [Reverted](https://github.com/dimagi/commcare-core/pull/1040/commits/52a35e1ec05aeb4077b882173efa8b157f216dfa ) to Simon's approach on initializing the remote instance from https://github.com/dimagi/commcare-core/pull/1038